src/errors/index.ts

```diff
@@ -21,7 +21,7 @@ const generateErrorMessage = (err: ApolloError) => {
     const errors = ((err.graphQLErrors || []) as readonly Error[])
       .concat(err.clientErrors || []);
     errors.forEach((error: Error) => {
-      const errorMessage = error
+      const errorMessage = error && error.message
         ? error.message
         : 'Error message not found.';
       message += `${errorMessage}\n`;
```

Comment:
```suggestion
      message += `${error.message || error}\n`;
```
Rather than using a placeholder string, let's try stringifying the `error` when `error.message` is not available?

Reply:
This broke one test @benjamn: https://github.com/apollographql/apollo-client/blob/issue-10394-websocket-errors/src/core/__tests__/QueryManager/index.ts#L374-L375

I think I'll keep this PR focused on addressing https://github.com/apollographql/apollo-client/issues/10394, so I've reverted this error formatting change and updated the types to make it clear that error is `unknown` (since non-spec-compliant servers can give us something like `result.errors = [null]`): https://github.com/apollographql/apollo-client/pull/10586/commits/e583b047dd1dbb0e27d710269eef3c9aa718ea10